### PR TITLE
Adding reference for `create_host_user_default_shell` role option

### PR DESCRIPTION
**Pull request by eriktate**
Documentation for #46539

Adding a reference for `create_host_user_default_shell` to the role spec.

### CODE REVIEW

File: docs/pages/includes/role-spec.mdx
Comment (ptgott) [inline docs/pages/includes/role-spec.mdx:114]:
```suggestion
    # Sets the default shell for auto-provisioned SSH users. An absolute path to a shell or a name
```

File: docs/pages/includes/role-spec.mdx
Comment (ptgott) [inline docs/pages/includes/role-spec.mdx:116]:
```suggestion
    # create_host_user_mode is not set to off.
```
Markdown isn't rendered in code snippet comments, so we can remove the backticks.

**Review by ptgott — APPROVED** [review]

**Review by rosstimothy — APPROVED** [review]
LGTM. Let's hold off on merging this until the functional PR with these changes lands.

**Pull request merged**
